irohad/torii/impl/command_service.cpp

```diff
@@ -75,68 +75,59 @@ namespace torii {
+        log_->warn("Found transaction {} in cache, ignoring", tx_hash.hex());
+        return;
+      }
 
-              // TODO: 08/08/2018 @muratovv remove duplication between Torii and
-              // ListTorii IR-1583
-              auto single_batch_result = shared_model::interface::
-                  TransactionBatch::createTransactionBatch(
-                      std::make_shared<shared_model::proto::Transaction>(
-                          std::move(iroha_tx.value)),
-                      shared_model::validation::
-                          DefaultSignedTransactionValidator());
-              single_batch_result.match(
-                  [this](const iroha::expected::Value<
-                         shared_model::interface::TransactionBatch> &value) {
-                    tx_processor_->batchHandle(value.value);
-                  },
-                  [this](const auto &err) {
-                    log_->warn("Transaction can't transformed to batch: {}",
-                               err.error);
-                  });
+      this->pushStatus(
+          "ToriiBatchProcessor",
+          tx_hash,
+          makeResponse(
+              tx_hash,
+              iroha::protocol::TxStatus::STATELESS_VALIDATION_SUCCESS));
+    });
+  }
 
-              this->pushStatus(
-                  "Torii",
-                  std::move(tx_hash),
-                  makeResponse(
-                      tx_hash,
-                      iroha::protocol::TxStatus::STATELESS_VALIDATION_SUCCESS));
-            },
-            [this, &request](const auto &error) {
-              // getting hash from invalid transaction
-              auto blobPayload =
-                  shared_model::proto::makeBlob(request.payload());
-              auto tx_hash =
-                  shared_model::crypto::DefaultHashProvider::makeHash(
-                      blobPayload);
-              log_->warn("Stateless invalid tx: {}, hash: {}",
-                         error.error,
-                         tx_hash.hex());
-
-              // setting response
-              auto response = makeResponse(
-                  tx_hash,
-                  iroha::protocol::TxStatus::STATELESS_VALIDATION_FAILED);
-              response.set_error_message(std::move(error.error));
-
-              this->pushStatus(
-                  "Torii", std::move(tx_hash), std::move(response));
-            });
+  void CommandService::Torii(const iroha::protocol::Transaction &request) {
+    auto tx_builder = shared_model::proto::TransportBuilder<
+        shared_model::proto::Transaction,
+        shared_model::validation::DefaultSignedTransactionValidator>();
+
+    auto batch_result = tx_builder.build(request) | [](const auto &iroha_tx) {
+      return shared_model::interface::TransactionBatch::createTransactionBatch(
+          std::make_shared<shared_model::proto::Transaction>(
+              iroha_tx),
+          shared_model::validation::DefaultSignedTransactionValidator());
```

Comment:
I suggest making the alias for DefaultSignedTransactionValidator inside the method and use it to avoid long read duplication.